### DISCUSSION TITLE
Sleeping time calculation

### DISCUSSION THREAD

**Discussion by pirilampr** [SilentVoid13/Templater#1027] [Help] [answered]
Hello, i'm trying to do a sleep calculation be automatic (using the tp.file.creation_date as time im going to sleep and tp.date.now as the time i wake up), but i can't make it to work. I started using de plugin some days ago, so i know little about it. (this is for a dream diary)

<%*
	var gt_sleep = tp.file.creation_date("HH[h]mm[m]");
	var w_up = tp.date.now("HH[h]mm[m]");
	var tf = date("24h00m");
	var diff = tf - gt_sleep + w_up;
	tR += `${diff}`
-%>

thats what i did, if someone can help me i would be very thankfull
(sorry for my bad english, it is not my firt language, i hope you could understand)

**Comment by welpdx** [comment, marked as answer]
You got the basic gist right! Since Obsidian uses [moment](https://momentjs.com/), we can use that with templater. 


```js
<%*  
var gt_sleep = moment(app.workspace.getActiveFile().stat.ctime);  
var w_up = moment();  

// below from stackoverflow: https://stackoverflow.com/a/29746202
// calculate total duration
var duration = moment.duration(w_up.diff(gt_sleep));
// duration in hours
var hours = parseInt(duration.asHours());
// duration in minutes
var minutes = parseInt(duration.asMinutes()) % 60;

var output = 'You slept: ' + hours + ' hour and ' + minutes + ' minutes.'
console.log(output);
tR += `${output}`  
-%>
```

<details>
           <summary style="font-size:20">Demo</summary>
           <img src="https://i.imgur.com/ihwO24T.gif" alt="demo" />
</details>

**Reply by pirilampr** [reply]
Wow it worked perfectly, thank you so much for the help :)

**Reply by welpdx** [reply]
np!
